tools/pre-commit.sh

```diff
@@ -1,28 +1,83 @@
 #!/usr/bin/env bash
+# Copy or link this script into .git/hooks/
+#   $ cp "`pwd`/tools/pre-commit.sh" "`pwd`/.git/hooks/pre-commit"
+# This script runs automatically in the project's root directory (parent of .git/).
 
-python -m black --check ./ > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m black ./
-    echo Some Python files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+EXIT_STATUS=0
+EXIT_EARLY=0
+MAX_FILE_SIZE=32 # Max JSON file size in Megabytes.
+PROJECT_ROOT=`git rev-parse --show-toplevel`
+
+# Determine command to use in order to collect
+# tracked files. Toggle between `dif` and `lstree`
+# using the `ARMORY_CI_TEST` environmental variable.
+# Example:
+#   $ ARMORY_CI_TEST=1 ./tools/pre-commit.sh
+ARMORY_CI_TEST="${ARMORY_COMMIT_HOOK_CI:-0}"
+
+TRACKED_FILES="git --no-pager diff --name-only"
+if [ "${ARMORY_CI_TEST}" -ne 0 ]; then
+    TRACKED_FILES="git --no-pager ls-files"
 fi
-set +e
-
-python -m tools.format_json --check > /dev/null 2>&1
-need_format=$?
-set -e
-if [ $need_format -ne 0 ]
-then
-    python -m tools.format_json
-    echo Some JSON files were formatted
-    echo You need to do git add and git commit again
-    exit $need_format
+[ -z "$TRACKED_FILES" ] && exit 0
+
+
+function CHECK_EXIT_STATUS ()
+{
+    if [ $1 -ne 0 ]; then
+        EXIT_STATUS=1
+        if [ $EXIT_EARLY -ne 0 ]; then
+          echo "🚨 Pre-commit hooks failed. Please fix the issues and re-run 'git add' and 'git commit' 🚑"
+          exit ${EXIT_STATUS}
+        fi
+    fi
+}
+
+
+pushd $PROJECT_ROOT > /dev/null || exit 1
+    ############
+    # Black
+    echo "⚫ Executing 'black' formatter..."
```

Comment:
This line should come after line 42 (as `black` isn't run if there are no target files.

Reply:
I added some verbosity to prior methods.